dissect/target/plugins/scrape/qfind.py

```diff
@@ -109,9 +133,21 @@ def qfind(
                 tmp[re.compile(re.escape(encoded_needle), re.IGNORECASE)] = _
             needle_lookup = tmp
 
+        if regex:
+            tmp = {}
+            for needle, _ in needle_lookup.items():
+                tmp[re.compile(needle, re.IGNORECASE if ignore_case else re.NOFLAG)] = _
```

Comment:
Does this work properly if the `ignore_case` branch just above this has already compiled the needle into a regex pattern? I get an error when trying to compile an already compiled pattern.